### ISSUE TITLE
Filter allowable imports in `pattern-where-python` to limit potential risk from untrusted rules

### ISSUE THREAD

**Issue by clintgibler** [semgrep/semgrep#478]
**Is your feature request related to a problem? Please describe.**
This isn't directly a problem or a bug, but a potential hardening step.

The `pattern-where-python` clause allows rule writers great flexibility in matching patterns by allowing them to vet what is matched with arbitrary Python code.

However, a malicious rule author who is able to trick a victim user to running a malicious rule could run arbitrary Python code on their machine.

Note that `pattern-where-python` is already gated by a separate, scary sounding flag. 

However, a user may already be using known, vetted, safe rules that use `pattern-where-python`; thus, they would not receive an additional warning when a new, potentially malicious rule is added to their existing rule set.

**Describe the solution you'd like**

There could potentially be a check when the YAML rules to run are first loaded that examines the code in the `pattern-where-python` clause and fails or logs a warning if "malicious" or potentially dangerous functionality is detected.

For example, if the Python code appears to be doing file I/O, making network requests, has nested `exec` or `eval` with strings of high entropy, uses `import`, etc.

* A quick, easy, imprecise solution would be to examine the passed string with regex whitelists or blacklists. 
* A better, and harder, solution would be to try to parse the string as Python code into an AST and vet that. However, preventing all potentially obfuscated bad things is not a solvable problem (*shakes fist at the Halting Problem* 🛑 )

Additionally, sgrep could consider warning users who are already using `pattern-where-python` when they add new, previously not run rules that leverage `pattern-where-python` clauses when the inner code is potentially dangerous (note: would add user friction).

**Describe alternatives you've considered**
We could try running sgrep in a sandbox (SELinux policy, Seatbelt, etc.) to limit the capabilities of what sgrep can do, but this is likely more effort than the security value it provides.

Note: users running sgrep in Docker will already have the impact (largely) limited.

**Additional context**
Note: we probably can't block importing any package, because that functionality can be quite useful, see: #412 "[SgrepAPI] ability to import python packages inside python-where clause"

This issue should likely be considered low priority, but I'm including it here for future tracking.

**Comment by aryx** [comment]
Do we still need this now that people can migrate most of the pattern-where-python to the safe metavariable-comparison of @mschwager ?

**Comment by ievans** [comment]
I don't think so. Closing and @mschwager or @clintgibler can re-open if they disagree
